api/views.py

```diff
@@ -9,6 +10,21 @@ class UpcomingWorkshops(ListAPIView):
     serializer_class = WorkshopSerializer
 
     def get_queryset(self):
-        if self.request.GET.get('status', None):
-            return Workshop.objects.filter(status=self.request.GET.get('status'))
-        return Workshop.objects.all()
+        params = self.request.GET
+        queryset = Workshop.objects.all()
+        if params.get('status', None):
+            try:
+                queryset = queryset.filter(status=params.get('status'))
+            except:
+                pass
+        if params.get('date_from'):
+            try:
+                queryset = queryset.filter(date__gte=dateutil.parser.parse(params.get('date_from')).date())
```

Comment:
You don't need to do any parse here. If the date is not proper the filter will return empty.